### PR TITLE
Bump klog v0.3.1 to v0.4.0

### PR DESCRIPTION
**Pull request by mytunguyen**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by ncdc** [comment]
Hi @mytunguyen, please try to rebase this when you have some spare time. Thanks!

**Comment by mytunguyen** [comment]
WIP while I figure out these tests :)

**Comment by vincepri** [comment]
@mytunguyen Newer klog has added a new flag, you should be able to update the testdata changing `clusterctl/main_integration_test.go` update flag to true

**Comment by mytunguyen** [comment]
> @mytunguyen Newer klog has added a new flag, you should be able to update the testdata changing `clusterctl/main_integration_test.go` update flag to true

perfect. thanks a bunch

File: cmd/clusterctl/main_integration_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ const (
 
 // run these tests with the flag "-update" to update the values stored in all of the golden files
 var (
-	update = flag.Bool("update", false, "update .golden files")
+	update = flag.Bool("update", true, "update .golden files")
```
Comment (vincepri) [inline cmd/clusterctl/main_integration_test.go]: This should be reverted before merging :)

Comment (mytunguyen) [reply cmd/clusterctl/main_integration_test.go]: Ohhhhhh I get it now

**Pull request merged**